README.md

```diff
@@ -1,4 +1,253 @@
+
+# for compatibility with csv_permissions 0.1.0
+#CSV_PERMISSIONS_RESOLVE_EVALUATORS = "csv_permissions.legacy.resolve_evaluators"
+
+```
+
+The following code will define some custom evaluators: 
+- `'if_monday'` grants all access on mondays.
+- `'all_caps'` grants access to all objects that have a `name` field containing
+    all uppercase.
+
+In `my_app.evaluators`:
+```python
+import datetime
+from typing import Optional
+
+from csv_permissions.types import Evaluator
+from csv_permissions.types import UnresolvedEvaluator
+
+
+def evaluate_if_monday(user, obj=None):
+    return datetime.datetime.today().weekday() == 0
+
+def evaluate_all_caps(user, obj=None):
+    if obj is None:
+        raise ValueError("'all_caps' cannot be used as a global permission.")
+    
+    try:
+        return obj.name.isupper()
+    except AttributeError:
+        return False
+     
+def resolve_evaluators(details: UnresolvedEvaluator) -> Optional[Evaluator]:
+    if details.evaluator_name == "if_monday":
+        return evaluate_if_monday
+
+    if details.evaluator_name == "all_caps":
+        if details.is_global != False:
+            raise ValueError("'all_caps' cannot be used as a global permission.")
+        return evaluate_if_monday
+
+    return None
+```
+
+* Note that evaluator names do not have to be static strings: you could implement
+    something that understood `'all_caps:True'` and `'all_caps:False'` for example
+
+### Unrecognised Permissions
+
+If `settings.CSV_PERMISSIONS_STRICT` is true then querying a permission
+(or `user_type`) that is not in the CSV will raise a `LookupError`.
+
+This is not set by default as it prevents the ability to use multiple
+authentication backends for permission checks. If you are using `csv_permissions`
+exclusively for permission checks then it can be helpful to catch typos.
+
+### Permission names
+
+By default `csv_permissions` will use the same permission name format as django: `<app label>.<action>_<model>`
+
+You can optionally set `settings.CSV_PERMISSIONS_RESOLVE_PERM_NAME` to the fully qualified name of a function to
+resolve permission names to whatever pattern you want.
+
+In `settings.py`:
+```python
+CSV_PERMISSIONS_RESOLVE_PERM_NAME = 'my_site.auth.resolve_rule_name'
+```
+
+In `my_site/auth.py`:
+```python
+from typing import Optional
+from typing import Type
+
+from django.apps import AppConfig
+from django.apps import apps
+from django.db.models import Model
+
+def resolve_rule_name(app_config: AppConfig, model: Optional[Type[Model]], action: str, is_global: bool) -> str:
+    # here's an implementation that is almost the same as django, but
+    # uses - as a separator instead of _ and .
+    #
+    # we also need to handle with the case where a permission has no associated model
+    if model is None:
+        f"{app_config.label}-{action}"
+    else:
+        return f"{app_config.label}-{action}-{model._meta.model_name}"
+
+```
```

Comment:
Added an extra para:

```
Note the handling of the case where a permission has no model.
Examples of this can be seen in the `report_outstanding` and `report_popularity`
permissions in the [sample CSV file](#the-csv-file).
```